std/concurrency.d

```diff
@@ -530,17 +586,19 @@ if (isSpawnable!(F, T))
  * Returns:
  *  A Tid representing the new thread.
  */
-Tid spawnLinked(F, T...)(F fn, T args)
+Tid spawnLinked(F, T...)(F fn, auto ref T args)
 if (isSpawnable!(F, T))
 {
+    import std.functional : forward;
     static assert(!hasLocalAliasing!(T), "Aliases to mutable thread-local data not allowed.");
-    return _spawn(true, fn, args);
+    static assert(!hasLocalReferencePassing!(F, T), "Passing by reference thread-local data is not allowed.");
+    return _spawn(true, fn, forward!args);
 }
 
 /*
  *
  */
-private Tid _spawn(F, T...)(bool linked, F fn, T args)
+private Tid _spawn(F, T...)(bool linked, F fn, auto ref T args)
```

Comment:
Should the `fn(args)` [call](https://github.com/dlang/phobos/pull/7124/files#diff-25bf2af9a45af72e9499b850a56d6f32R612) (L612 in changed) from `exec` also forward ( `fn(forward!args);` )?

Reply:
oh true! I'm going to fix